sw/device/lib/dif/dif_plic.c

```diff
@@ -179,13 +204,55 @@ static ptrdiff_t plic_priority_reg_offset(dif_plic_irq_id_t irq) {
   return RV_PLIC_PRIO0_REG_OFFSET + offset;
 }
 
+/**
+ * Reset the requested PLIC peripheral.
+ *
+ * This function resets all the relevant PLIC registers, apart from the CC
+ * register. There is no reliable way of knowing the ID of an IRQ that has
+ * claimed the CC register, so we assume that the previous "owner" of the
+ * resource has cleared/completed the CC access.
+ */
+static void plic_reset(const dif_plic_t *plic) {
+  // Clear all of Interrupt Enable and Level/Edge registers.
+  for (int i = 0; i < plic_num_irq_reg(); ++i) {
+    ptrdiff_t offset = RV_PLIC_IE00_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+
+    offset = RV_PLIC_LE0_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear all of the priority registers.
+  for (int i = 0; i < plic_num_priority_reg(); ++i) {
+    ptrdiff_t offset = RV_PLIC_PRIO0_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear all of the target threshold registers.
+  for (dif_plic_target_t target = kDifPlicTargetIbex0;
+       target <= kDifPlicTargetLast; ++target) {
+    ptrdiff_t offset = plic_target_reg_offsets[target].threshold;
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear software interrupt pending register.
+  mmio_region_write32(plic->base_addr, RV_PLIC_MSIP0_REG_OFFSET, 0);
+
+  // TODO - is there any way to ensure that all CC claimed operations have been
```

Comment:
Can you file an issue for this so we don't lose track of it?

Reply:
OOoops, forgot to delete this comment. I have added description for this function? Please let me know if you would like me to rephrase the description, or add anything?

I have deleted the TODO now and will push shortly.